meson.build

```diff
@@ -1,86 +1,177 @@
+version_cfg.set('LCMS2_VERSION_MINOR', version_components.get(1, 0))
+version_cfg.set('LCMS2_VERSION_MICRO', version_components.get(2, 0))
+version_cfg.set_quoted('LCMS2_VERSION', meson.project_version())
 
 cc = meson.get_compiler('c')
 
-lcms2_srcs = [
-  'src/cmsalpha.c',
-  'src/cmscgats.c',
-  'src/cmserr.c',
-  'src/cmsgmt.c',
-  'src/cmsintrp.c',
-  'src/cmsio1.c',
-  'src/cmsmd5.c',
-  'src/cmsnamed.c',
-  'src/cmspack.c',
-  'src/cmsplugin.c',
-  'src/cmssamp.c',
-  'src/cmstypes.c',
-  'src/cmswtpnt.c',
-  'src/cmscam02.c',
-  'src/cmscnvrt.c',
-  'src/cmsgamma.c',
-  'src/cmshalf.c',
-  'src/cmsio0.c',
-  'src/cmslut.c',
-  'src/cmsmtrx.c',
-  'src/cmsopt.c',
-  'src/cmspcs.c',
-  'src/cmsps2.c',
-  'src/cmssm.c',
-  'src/cmsvirt.c',
-  'src/cmsxform.c',
-]
-
-install_headers([
-  'include/lcms2.h',
-  'include/lcms2_plugin.h'
-])
-
-inc_dirs = include_directories('include')
-
-deps = [
-  cc.find_library('m', required: false),
-  dependency('threads'),
-]
-
-cargs = ['-DHasTHREADS=1']
+cargs = []
+
+if cc.has_function_attribute('visibility:hidden')
+  cargs += '-DHAVE_FUNC_ATTRIBUTE_VISIBILITY=1'
+endif
+
+if host_machine.endian() == 'big'
+  cargs = '-DWORDS_BIGENDIAN=1'
+endif
+
+if (
+  not cc.has_header_symbol('features.h', '_POSIX_C_SOURCE')
+  and cc.get_define('_POSIX_C_SOURCE') == ''
+)
+  cargs += ['-D_POSIX_C_SOURCE=199506L']
+endif
 
 # Check for threadsafe variants of gmtime
-if cc.has_function('gmtime_r', prefix: '#include <time.h>')
+if cc.has_header_symbol(
+  'time.h',
+  'gmtime_r',
+  args: cargs,
+)
   cargs += '-DHAVE_GMTIME_R=1'
-elif cc.has_function('gmtime_s', prefix: '#include <time.h>')
-  cargs += '-DHAVE_GMTIME_S=1'
-endif
+elif cc.has_header_symbol('time.h', 'gmtime_s')
+  if cc.links(
+    '''
+    #include <time.h>
 
-if cc.has_function_attribute('visibility:hidden')
-  cargs +='-DHAVE_FUNC_ATTRIBUTE_VISIBILITY=1'
+    int main() {
+      time_t t;
+      struct tm m;
+      gmtime_s(&m, &t);
+      return 0;
+    }
+    ''',
+    name: 'gmtime_s can be used',
+  )
+    cargs += '-DHAVE_GMTIME_S=1'
+  endif
 endif
 
-if host_machine.system() == 'windows'
-  win = import('windows')
-  lcms2_srcs += win.compile_resources('Projects/VC2019/lcms2.rc')
-  cargs += '-DCMS_DLL_BUILD=1'
+jpeg_dep = dependency(
+  'libjpeg',
+  'jpeg',
+  required: get_option('jpeg'),
+)
+
+if jpeg_dep.found()
+  if (
+    cc.check_header('jconfig.h', dependencies: jpeg_dep)
+    and cc.check_header('jerror.h', dependencies: jpeg_dep)
+    and cc.check_header(
+      'jmorecfg.h',
+      prefix: '#include <jconfig.h>',
+      dependencies: jpeg_dep,
+    )
+    and cc.check_header(
+      'jpeglib.h',
+      prefix: '''
+    #include <stdio.h>
+    #include <stdlib.h>
+    ''',
+      dependencies: jpeg_dep,
+    )
+    and cc.compiles(
+      '''
+    #include <stdio.h>
+    #include <stdlib.h>
+    #include <jpeglib.h>
+    #if JPEG_LIB_VERSION < 62
+    #error IJG JPEG library must be version 6b or newer!
+    #endif
```

Comment:
This is... a LOT of tests just to check the version.

You will never find an IJG JPEG that old anyway, since they don't provide pkg-config until 9c. If this PR was updated to be compatible with CMake's totally unofficial FindJPEG.cmake, then that extracts the version from jpeglib.h or jconfig.h's `JPEG_LIB_VERSION` and sets it to the cmake variable JPEG_VERSION, which means that Meson will see that too.

No matter how you look at it, `jpeg_dep.version()` will provide a suitable version.

...

But also are there really people who still use the IJG version? Wow.